### PR TITLE
Deprecate OAuth 1.0 and /identity/connect/register endpoint 

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request

Deprecate OAuth 1.0 and /identity/connect/register endpoint 

**Related issues**: https://github.com/wso2/product-is/issues/8564, https://github.com/wso2/product-is/issues/8614

### When should this PR be merged
This PR should be merged after https://github.com/wso2/carbon-identity-framework/pull/3027

### Follow up actions
After merging https://github.com/wso2/carbon-identity-framework/pull/3027 the released identity-framework version should be updated before merging this.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp [hunk 1]
```diff
@@ -444,11 +445,18 @@
                                 <tr>
                                     <td class="leftCol-med"><fmt:message key='oauth.version'/><span
                                             class="required">*</span></td>
+                                    <%
+                                    boolean isOAuth1Enabled = Boolean.parseBoolean(IdentityUtil.getProperty("LegacyFeatures.EnableOAuth1"));
+                                    if (!isOAuth1Enabled) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp:450]: shouldn't we prevent displaying the input field if if isOAuth1Enabled not enabled?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java [hunk 2]
```diff
@@ -41,19 +43,32 @@
...
+    public static final String ENABLE_OAUTH1_CONFIG = "LegacyFeatures.EnableOAuth1";
+
+    protected void activate(ComponentContext context) throws IdentityOAuthClientException {
+
+        boolean isOAuth1Enabled = Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_OAUTH1_CONFIG));
+        if (!isOAuth1Enabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("OAuth 1.0 Endpoint is disabled.");
+            }
+            throw new IdentityOAuthClientException(FORBIDDEN.getErrorCode(), "OAuth 1.0 is not supported");
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java:69]: this is not needed. We don't need to disable the OSGi service component.

File: components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp (hunk 1 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp]: Corrected https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/e5dca5f8761be050156782479f65a2d611f76780

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java (hunk 2 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java]: Corrected https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/e5dca5f8761be050156782479f65a2d611f76780

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/OAuthServiceComponent.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/e5dca5f8761be050156782479f65a2d611f76780

**Comment by emswbandara** [comment]
@GANGANI can you check why travis is failing?

File: components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp [hunk 3]
```diff
@@ -444,11 +445,20 @@
...
                                     <td class="leftCol-med"><fmt:message key='oauth.version'/><span
                                             class="required">*</span></td>
-                                    <td><input id="oauthVersion10a" name="oauthVersion" type="radio"
-                                               value="<%=OAuthConstants.OAuthVersions.VERSION_1A%>"/>1.0a
+                                    <td>
+                                    <%
+                                    boolean OAuth1FieldDisabled = false;
+                                    boolean isOAuth1Enabled = Boolean.parseBoolean(IdentityUtil.getProperty("LegacyFeatures.EnableOAuth1"));
+                                    if (isOAuth1Enabled) {
+                                        OAuth1FieldDisabled = true;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp:453]: why we set OAuth1FieldDisabled to true when OAuth1 is enabled?

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/1f613f565281710a85847663e9e82f1ed21e34db

File: components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java [hunk 4]
```diff
@@ -54,9 +54,14 @@
...
             if (log.isDebugEnabled()) {
-                log.debug("Identity Connect DCR Endpoint is disabled.");
+                log.debug("Identity Connect DCR endpoint was deprecated. To enable the DCR API endpoint " +
+                        "add the following config to deployment.toml file." +
+                        "[legacy_features]" +
+                        "enable_identity_connect_dcr = true");
             }
-            throw IdentityException.error(DCRException.class, ErrorCodes.FORBIDDEN.toString());
+            String errorMessage =
+                    "/identity/connect/register API was deprecated. Refer /api/identity/oauth2/dcr endpoint";
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java:63]: let's not hard-code the /api/identity/oauth2/dcr. Instead let's add "Refer the WSO2 official documentation for more information` <link>`"

File: components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java [hunk 5]
```diff
@@ -54,9 +54,14 @@
         boolean isIdentityConnectDCREnabled = Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_OIDC_DCR_CONFIG));
         if (!isIdentityConnectDCREnabled) {
             if (log.isDebugEnabled()) {
-                log.debug("Identity Connect DCR Endpoint is disabled.");
+                log.debug("Identity Connect DCR endpoint was deprecated. To enable the DCR API endpoint " +
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java:57]: change as "Identity Connect DCR endpoint is deprecated and hence disabled by default in the server. To enable it add the following config..."

File: components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java (hunk 4 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/515007e26b0cd144ab864d90dfd2e76cfc8f9c2c

File: components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java (hunk 5 above)
Comment (GANGANI) [reply components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java:57]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/515007e26b0cd144ab864d90dfd2e76cfc8f9c2c

**Comment by emswbandara** [comment]
@GANGANI let's update the framework version once the framework PR is merged.

File: components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java [hunk 6]
```diff
@@ -44,6 +48,24 @@
...
+        boolean isIdentityConnectDCREnabled =
+                IdentityUtil.isLegacyFeatureEnabled(OIDCDCRConstants.OIDC_DCR_ID, OIDCDCRConstants.OIDC_DCR_VERSION);
+
+        if (!isIdentityConnectDCREnabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("Identity Connect DCR endpoint was deprecated. To enable the DCR API endpoint " +
+                        "add the following config to deployment.toml file." +
+                        "[[legacy_feature]]" +
+                        "id = identity/connect/dcr" +
+                        "enable = true");
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java:61]:
```suggestion
                        "[[legacy_feature]] \n" +
                        "id = identity/connect/dcr \n" +
                        "enable = true");
```

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oidc.dcr/src/main/java/org/wso2/carbon/identity/oidc/dcr/processor/OIDCDCRProcessor.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407/commits/54221d9ca83bffb0964b332d92e9573fd1c053bf

File: components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/ErrorCodes.java [hunk 7]
```diff
@@ -22,5 +22,7 @@
  */
 public enum ErrorCodes {
     META_DATA_VALIDATION_FAILED,
-    BAD_REQUEST;
+    BAD_REQUEST,
+    FORBIDDEN;
+
```
Comment (tharindu-bandara) [inline components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/ErrorCodes.java:27]: Redundant new line

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/ErrorCodes.java:27]: These code changes were reverted recently. I'll fix this with the new PR.

**Pull request merged**